google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentChange.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A DocumentChange represents a change to the documents matching a query. It contains the document
+ * affected and a the type of change that occurred (added, modifed, or removed).
+ */
+public class DocumentChange {
```

Comment:
Does it make sense to make this `@AutoValue`? If you'd rather keep it Android-like, it's fine with me.

Reply:
I think it's better for us in the long run if we share code where we can. It should also make the chunks of code that need thorough reviewing a lot smaller.